src/Products/ZCatalog/tests/test_query.py

```diff
@@ -98,3 +98,25 @@ def test_options_string(self):
                           self._makeOne,
                           request, 'path',
                           ('query', 'operator'))
+
+    def test_get_string_with_underscores(self):
+        request = {
+            'extra_path': 'foo',
+            'extra_path_level': 0,
+            'extra_path_operator': 'and',
+        }
+        parser = self._makeOne(request,
+                               'extra_path',
+                               ('query', 'level', 'operator'))
+
+        self.assertEqual(parser.get('keys'), ['foo'])
+        self.assertEqual(parser.get('level'), 0)
+        self.assertEqual(parser.get('operator'), 'and')
+
+    def test_operator_string_with_underscores(self):
+        request = {'extra_path': 'foo', 'extra_path_operator': 'bar'}
+        self.assertRaises(ValueError,
+                          self._makeOne,
+                          request, 'extra_path',
+                          ('query', 'operator'),
+                          ('or', 'and'))
```

Comment:
```suggestion
        self.assertRaises(ValueError):
            self._makeOne(
                request, 'extra_path', ('query', 'operator'), ('or', 'and'))
```

I think this is easier to read.

Reply:
I agree on the use of the context manager (note that a with is missing):
```python
with self.assertRaises(ValueError):
    ...
```